server.js

```diff
@@ -16,8 +19,59 @@ const welcomeMessage = {
 //Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+app.get("/", function (req, res) {
+  res.sendFile(__dirname + "/index.html");
+});
+
+// read all messages
+app.get("/messages", function(req,res) {
+  res.json(messages);
+})
+
+// read a message by id
+app.get("/messages/:id", function(req, res) {
+  const messageId = req.params.id;
+  const findId = messages.find((message) => { return message.id === Number(messageId) });
+    if (findId) {
```

Comment:
Tiny formatting error where "if" isn't lined up with const but don't worry about that for now!